pkg/skaffold/runner/runner_test.go

```diff
@@ -159,6 +213,38 @@ func createRunner(t *testutil.T, testBench *TestBench) *SkaffoldRunner {
 	runner.Syncer = testBench
 	runner.Tester = testBench
 	runner.Deployer = testBench
+	runner.Listener = testBench
+
+	testBench.devLoop = func() error {
```

Comment:
It feels super strange that we have to reimplement that part of the runner. It doesn't feel like we're testing the actual code. (Maybe it wasn't the case before either)

Reply:
yeah, this was the case before as well, TestBench is a big mock. it is very hard to see what is the actual part of production code that is tested.

Reply:
I've just checked and it looks like `TestBench` is implementing `Builder`, `Deployer`, `Tester`... but **didn't implement anything from the `SkaffoldRunner`**. It's basically a list of mocks that record the interactions of a `SkaffoldRunner` with the other components.
That doesn't seem to be the case anymore.

Reply:
yeah it's weird. I can't call `runner.Dev()` or `runner.WatchForChanges()` in the tests because those will run forever; so this was my way of "simulating" the dev loop by "mapping" out the general flow of it.

I'm not sure I totally follow what you're saying though....it sounds like you're proposing that we rewrite the `TestBench` to mock out the methods defined on the `SkaffoldRunner` interface? I don't totally disagree with you (I don't really like the way this test is structure to begin with) but it would be a pretty big change. what I did here was the most minimal way I found to retrofit the old tests to the new structure.

Reply:
What I'm saying is that `runner_test.go` tries to test the code in `runner.go`, so, the SkaffoldRunner. If feels weird to reimplement the devloop part in tests because then, what are we testing?
The idea of this test was to mock the Builder, the Deployer, the Tester..., group then under the name TestBench and see how SkaffoldRunner interacts with them.

Reply:
ok I see what you mean. I took a stab at moving the dev loop out to a method on the `SkaffoldRunner`, and then calling that from the tests, with a tiny wrapper that calls the file monitor. this is the same thing the actual `Listener` does whenever it receives a trigger, so it should be actually testing the correct thing now. have a look at the latest commit and tell me what you think